ui/components/src/context/PersesThemeProvider.tsx

```diff
@@ -0,0 +1,195 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import { useTheme, ThemeOptions as MaterialThemeOptions } from '@mui/material';
+import merge from 'lodash/merge';
+import { EChartsTheme, PersesChartsTheme } from '../model';
+import { ChartsThemeProvider } from './ChartsThemeProvider';
+
+export interface PersesThemeProviderProps {
+  themeName: string;
+  muiTheme?: MaterialThemeOptions;
+  themeOverrides?: EChartsTheme;
+  children?: React.ReactNode;
+}
+
+export function PersesThemeProvider(props: PersesThemeProviderProps) {
```

Comment:
Yeah it wouldn't work in the app since we need to import it in other apps besides Perses, but I agree, having it in components is a little weird--maybe @perses-dev/core would be better? (any thoughts @LukeTillman)

The way I named the providers is also causing confusion, see my [response here](https://github.com/perses/perses/pull/462#discussion_r900153326).

Reply:
And having a package just for the theme might be too small. I do think eventually having a new package for @perses-dev/charts would make sense. If I refactored the Tooltip in LineChart, it wouldn't need to have @mui/material as a dependency so if Prom wanted to pull in LineChart, it would only be taking on echarts as a new dependency.

Reply:
> it wouldn't need to have @mui/material as a dependency so if Prom wanted to pull in LineChart

That's indeed interesting if the package we provide has the minimum of dependency required !